src/conversions.cpp

```diff
@@ -0,0 +1,127 @@
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+ */
+
+#include <moveit/drake/conversions.hpp>
+namespace moveit::drake
+{
+using ::drake::geometry::SceneGraph;
+using ::drake::multibody::AddMultibodyPlantSceneGraph;
+using ::drake::multibody::MultibodyPlant;
+using ::drake::multibody::Parser;
+using ::drake::systems::DiagramBuilder;
+
+[[nodiscard]] ::drake::trajectories::PiecewisePolynomial<double>
+getPiecewisePolynomial(const ::robot_trajectory::RobotTrajectory& robot_trajectory,
+                       const moveit::core::JointModelGroup* group)
+{
+  std::vector<double> breaks;
+  breaks.reserve(robot_trajectory.getWayPointCount());
+  std::vector<Eigen::MatrixXd> samples;
+  samples.reserve(robot_trajectory.getWayPointCount());
+
+  // Create samples & breaks
+  for (std::size_t i = 0; i < robot_trajectory.getWayPointCount(); ++i)
+  {
+    const auto& state = robot_trajectory.getWayPoint(i);
+    Eigen::VectorXd position(state.getVariableCount());
+    state.copyJointGroupPositions(group, position);
+    samples.emplace_back(position);
+    breaks.emplace_back(robot_trajectory.getWayPointDurationFromStart(i));
+  }
+
+  // Create a piecewise polynomial trajectory
+  return ::drake::trajectories::PiecewisePolynomial<double>::FirstOrderHold(breaks, samples);
+}
+
+void getRobotTrajectory(const ::drake::trajectories::Trajectory<double>& drake_trajectory, const double delta_t,
+                        std::shared_ptr<::robot_trajectory::RobotTrajectory>& moveit_trajectory)
+{
+  // Reset output trajectory
+  moveit_trajectory->clear();
+
+  // Get the start and end times of the piecewise polynomial
+  double t_prev = 0.0;
+  const auto num_pts = static_cast<size_t>(std::ceil(drake_trajectory.end_time() / delta_t) + 1);
+
+  for (unsigned int i = 0; i < num_pts; ++i)
+  {
+    const auto t_scale = static_cast<double>(i) / static_cast<double>(num_pts - 1);
+    const auto t = std::min(t_scale, 1.0) * drake_trajectory.end_time();
+    const auto pos_val = drake_trajectory.value(t);
+    const auto vel_val = drake_trajectory.EvalDerivative(t);
+    const auto waypoint = std::make_shared<moveit::core::RobotState>(moveit_trajectory->getRobotModel());
+    const auto active_joints = moveit_trajectory->getGroup()->getActiveJointModels();
+    for (size_t joint_index = 0; joint_index < active_joints.size(); joint_index++)
+    {
+      waypoint->setJointPositions(active_joints[joint_index], &pos_val(joint_index));
+      waypoint->setJointVelocities(active_joints[joint_index], &vel_val(joint_index));
+    }
+
+    moveit_trajectory->addSuffixWayPoint(waypoint, t - t_prev);
+    t_prev = t;
+  }
+}
+
+[[nodiscard]] Eigen::VectorXd getJointPositions(const moveit::core::RobotState& moveit_state,
```

Comment:
Again, do you think this function can be name better to clarify which way the conversions are happening?